### PR TITLE
EWPP-4157: Add cron function to delete old contact messages.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/Plugin/views/field/ContactMessageRemainingDays.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+class ContactMessageRemainingDays extends FieldPluginBase {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory) {
```
Comment (nagyad) [inline src/Plugin/views/field/ContactMessageRemainingDays.php:28]: /**
   * {@inheritdoc}
   */

File: tests/src/Functional/RemainingDaysTest.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace Drupal\Tests\oe_contact_forms\Functional;
+
+use Drupal\contact\Entity\ContactForm;
+use Drupal\contact\Entity\Message;
+use Drupal\Tests\BrowserHtmlDebugTrait;
+use Drupal\Tests\views\Functional\ViewTestBase;
+
+/**
+ * Tests the contact link field.
```
Comment (nagyad) [inline tests/src/Functional/RemainingDaysTest.php:11]: Needs to be fixed.

File: tests/src/Functional/RemainingDaysTest.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ * @group contact
+ * @see \Drupal\oe_contact_forms\Plugin\views\field\ContactMessageRemainingDays.
+ */
+class RemainingDaysTest extends ViewTestBase {
+
+  use BrowserHtmlDebugTrait;
```
Comment (nagyad) [inline tests/src/Functional/RemainingDaysTest.php:18]: This is not needed.

File: oe_contact_forms.module [hunk 4]
```diff
@@ -996,3 +996,73 @@ function _oe_contact_forms_contact_language_values(): array {
...
+    $storage = \Drupal::entityTypeManager()->getStorage('contact_message');
+    $query = $storage->getQuery()
+      ->condition('created', $date, '<')
+      ->accessCheck()
+      ->range(0, 100)
+      ->execute();
+    if ($query) {
+      $storage->delete($storage->loadMultiple($query));
+    }
+  }
```
Comment (nagyad) [inline oe_contact_forms.module:1043]: This code is not tested.

File: oe_contact_forms.module [hunk 5]
```diff
@@ -996,3 +996,73 @@ function _oe_contact_forms_contact_language_values(): array {
...
+ */
+function oe_contact_forms_form_contact_storage_settings_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['auto_delete'] = [
+    '#type' => 'number',
+    '#title' => t('Auto delete'),
+    '#description' => t('Set the number of days after which the contact messages will be automatically deleted. Set to 0 to disable.'),
+    '#min' => 0,
+    '#default_value' => \Drupal::config('contact.settings')->get('auto_delete') ?? 0,
+  ];
+  $form['#submit'][] = '_oe_contact_forms_contact_storage_settings_submit';
```
Comment (nagyad) [inline oe_contact_forms.module:1011]: The form field is not tested.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — APPROVED** [review]

File: src/Plugin/views/field/ContactMessageRemainingDays.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    /** @var \Drupal\contact\Entity\Message $entity */
+    $entity = $this->getEntity($values);
+    $created_date = $entity->get('created')->value;
+    $auto_delete_days = $this->configFactory->get('contact.settings')->get('auto_delete');
+    if ($auto_delete_days === NULL) {
+      return [];
+    }
+    else {
+      $remaining_days = $auto_delete_days - floor((time() - $created_date) / (60 * 60 * 24));
+    }
```
Comment (22Alexandra) [inline src/Plugin/views/field/ContactMessageRemainingDays.php:61]: Do we need the `else` here?
```suggestion
    $remaining_days = $auto_delete_days - floor((time() - $created_date) / (60 * 60 * 24));
```

File: src/Plugin/views/field/ContactMessageRemainingDays.php [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+    $remaining_days = (int) $remaining_days < 0 ? 0 : (int) $remaining_days;
```
Comment (22Alexandra) [inline src/Plugin/views/field/ContactMessageRemainingDays.php:63]:
```suggestion
    $remaining_days = max((int)$remaining_days, 0);
```

**Review by upchuk — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**